cmake/kconfig.cmake

```diff
@@ -103,20 +103,20 @@ endforeach()
 # Create a new .config if it does not exists, or if the checksum of
 # the dependencies has changed
 set(merge_config_files_checksum_file ${PROJECT_BINARY_DIR}/.cmake.dotconfig.checksum)
-set(CREATE_NEW_DOTCONFIG "")
-if(NOT EXISTS ${DOTCONFIG})
-  set(CREATE_NEW_DOTCONFIG 1)
-else()
+set(CREATE_NEW_DOTCONFIG 1)
+# Check if the checksum file exists too before trying to open it, though it
+# should under normal circumstances
```

Comment:
Would happen if something goes wrong between writing `zephyr/.config` and calculating the checksum. Probably best to just retry in that case.

Reply:
Calculating the checksum is done before writing zephyr/.config AFAICT. So this condition would indicate an internal error / unreachable code situation. Internal errors / unreachable code should cause a failure as close to the root as possible to ease debugging (fail-fast).

Reply:
It is written after writing `zephyr/.config` though. Another fix would be to write the checksum file before writing `zephyr/.config` I think, so that `zephyr/.config` marks stuff as done.

Reply:
I added on that part as well.

Reply:
"It is written after writing zephyr/.config though."

My mistake.